### PR TITLE
restore server.ts

### PR DESCRIPTION
**Pull request by kpawelczak**
closes: [CXSPA-5855](https://jira.tools.sap/browse/CXSPA-5855)

- updateServerFiles(): This is a schematic rule that calls another rule called removeServer.

- removeServer(). It traverses the file tree and identifies two paths 'server.ts' and 'server.ts.bak'. If both paths are found, it deletes the original server file and renames a backup file to match the original name.

### CODE REVIEW

File: projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files.ts [hunk 1]
```diff
@@ -4,8 +4,42 @@
...
 
 export function updateServerFiles(): Rule {
-  return chain([]);
+  return chain([removeServer]);
+}
+
+function removeServer(): Rule {
+  return (tree: Tree, _context: SchematicContext) => {
+    let serverPath;
+    let serverBakPath;
```
Comment (pawelfras) [inline projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files.ts]: let's add typing as in the case of assigning the value inside the `visit` loop the type won't be inferred:
image

```suggestion
    let serverPath: Path | undefined;
    let serverBakPath: Path | undefined;
```

File: projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files.ts [hunk 2]
```diff
@@ -4,8 +4,42 @@
...
+    tree.visit((filePath: Path) => {
+      const fileName = filePath.replace(/^.*[\\/]/, '');
+      if (fileName === SERVER_FILENAME) {
+        serverPath = filePath;
+      }
+
+      if (fileName === SERVER_BAK_FILENAME) {
+        serverBakPath = filePath;
+      }
+    });
```
Comment (pawelfras) [inline projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files.ts:40]: `nitpick:` Would be great to break the loop as soon as we find both paths. If there is no easy way to do it, let's leave it as it is.

File: projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files_spec.ts [hunk 3]
```diff
@@ -77,4 +78,20 @@ describe('Update SSR', () => {
       tree
     );
   });
+
+  describe('updateServerFile', () => {
+    it('should rename server.bak to server.ts', async () => {
```
Comment (pawelfras) [inline projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files_spec.ts]:
```suggestion
    it('should restore server.ts based on the server.ts.bak and remove server.ts.bak file', async () => {
```

File: projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files_spec.ts [hunk 4]
```diff
@@ -77,4 +78,20 @@ describe('Update SSR', () => {
...
+      tree.create(serverBakPath, 'testing');
+      expect(tree.exists(serverBakPath)).toBeTruthy();
+      tree = await updateSsrSchematicRunner.runSchematic(
+        'update-ssr',
+        { name: 'schematics-test' },
+        tree
+      );
+
+      expect(tree.exists(serverBakPath)).toBeFalsy();
+      expect(tree.exists(SERVER_FILENAME)).toBeTruthy();
```
Comment (pawelfras) [inline projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files_spec.ts:103]: would be nice to test whether the `server.ts` file is different after restoration.

File: projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files_spec.ts (hunk 3 above)
Comment (kpawelczak) [reply projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files_spec.ts]: replaced

File: projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files.ts (hunk 2 above)
Comment (kpawelczak) [reply projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files.ts:40]: I've already thought about it and couldn't find an easy solution.

File: projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files.ts (hunk 1 above)
Comment (kpawelczak) [reply projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files.ts]: I've added types

File: projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files_spec.ts (hunk 4 above)
Comment (kpawelczak) [reply projects/schematics/src/migrations/mechanism/update-ssr/update-ssr-files_spec.ts:103]: I've added one more expect. It checks whether the restored server.ts has the same content as server.ts.bak.

**Review by pawelfras — APPROVED** [review]
LGTM

**Pull request merged**